tests/bluetooth/audio/cap_commander/uut/bap_broadcast_assistant.c

```diff
@@ -83,6 +93,12 @@ int bt_bap_broadcast_assistant_add_src(struct bt_conn *conn,
 	struct bt_bap_scan_delegator_recv_state state;
 	struct bt_bap_broadcast_assistant_cb *listener, *next;
 
+	if (add_src_skip != 0) {
+		add_src_skip--;
+
+		return 0;
+	}
```

Comment:
Do I understand this feature correctly: 

We can set a value, `add_src_skip`, that effectively delays the response from the server indefinitely, thus causing the procedure to never finish on it's own, which allows us to cancel it? 

If so, then it's a creative solution. Could this have been done with the mocked k_work implementation we have? 

Two comments about the design of this:
1) In some ways, it's a bit odd that we skip the first N callbacks, and not the last N callbacks, but the result will be the same
2) This is not a very scalable solution if we want to do the commander cancel test for the remaining CAP commander procedures. On the top of my head, I don't have a general solution for this though.

Reply:
Yes, you understand it correctly.

1. It is a bit odd, but doing it the otherway around would mean rewriting other tests
2. I couldn't come with a better solution either :(

Reply:
Spend some more time for a better solution, but this is still the best I can come up with for now